src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -127,18 +127,56 @@ public static bool TryCreateNativeIntegerType(INamedTypeSymbol symbol, out TypeS
 
 #if !CODE_STYLE
 
+#nullable enable
             public override TypeSyntax VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
             {
-                // TODO(https://github.com/dotnet/roslyn/issues/39865): generate the calling convention once exposed through the API
+                FunctionPointerCallingConventionSyntax? callingConventionSyntax = null;
+                if (symbol.Signature.CallingConvention != System.Reflection.Metadata.SignatureCallingConvention.Default)
+                {
+                    SeparatedSyntaxList<FunctionPointerUnmanagedCallingConventionSyntax> conventionsList = default;
+                    switch (symbol.Signature.CallingConvention)
+                    {
+                        case System.Reflection.Metadata.SignatureCallingConvention.CDecl:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Cdecl") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.StdCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Stdcall") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.ThisCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Thiscall") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.FastCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Fastcall") });
+                            break;
+
+                        default:
```

Comment:
Consider case System.Reflection.Metadata.SignatureCallingConvention.Unmanaged here, and then adding a throw UnexpectedValue() for the default case if that enum ever changes.